### PR TITLE
Added utility classes and simple cluster example

### PR DESCRIPTION
**Pull request by JHoelzel**

### CODE REVIEW

**Comment by JHoelzel** [comment]
also addet apt key update & fixed some travis warnings

File: manifests/utility/database.pp [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Sample Usage:
+#   mariadb::utility::database { "production dbname" :
+#       dbname => 'dbname',
+#       charset => 'utf8',
+#       collate => 'utf8_general_ci'
+#       ensure => 'present',
+#   }
+define mariadb::utility::database($dbname, $charset = 'utf8', $collate = 'utf8_general_ci', $ensure = 'present' ) {
```
Comment (edestecd) [inline manifests/utility/database.pp]: What is the advantage of using these utility defined types vs just using the resources from the mysql module directly? What problem is this solving for you?

File: manifests/repo/percona/apt.pp [hunk 2]
```diff
@@ -1,16 +1,26 @@
 # apt.pp
 # Manage the percona apt repo.
+# with different keys for ubuntu and debian
 #
```
Comment (edestecd) [inline manifests/repo/percona/apt.pp]: Can you fix the lint/spec errors?

File: manifests/utility/database.pp (hunk 1 above)
Comment (JHoelzel) [inline manifests/utility/database.pp]: I see this as neccesairy for multiple reason;
- usuability. One would expect them here. You can pretty much setup the Server in one a oneliner, why not ensure that users dbs and grants are also included.
- One does not necessairy need to know about the core module to use it
- Furthermore there should be added validation anyway. So far i have added just ensure but ther could be more
- finally when i see a project like this, i think we agree that the main reason for doing this is getting the show on the road quickly. this makes it easy and its not like its including megabytes of unneccesairy code.

i can totally see though if you see this as clutter, in that case just replace it with core module calls =)

Comment (edestecd) [inline manifests/utility/database.pp]: We take in hashes already to pass on to the underlying mysql module here: https://github.com/edestecd/puppet-mariadb/blob/master/manifests/cluster.pp#L78
I think that solves the "one liner", or less code approach. More documentation would be nice, as maybe thats not clear.

I worry about maintaining such wrappers. Every time the underlying types change in the mysql module, we would be constantly updating parameters to our wrappers. I think that is harder to maintain. You don't have that problem passing through hashes. I think just using the underlying types as they are and contributing to them in that module is the best approach. I do not wish to provide that functionality as I think the mysql module does it well.

**Comment by edestecd** [comment]
Can you break this up into a few Pull Requests?  
I would like to see the simple percona repo fix go in right away and I would like you to get credit for it. We can work out the details on the documentation and types in anther slower paced PR...

**Comment by edestecd** [comment]
Can we get a rebase?

**Pull request closed without merging**